tests/regression/test_constant.py

```diff
@@ -231,3 +233,33 @@ class CustomConstant(Constant):
 
     assert const2.count() == const1.count() + 1
     assert const3.count() == const1.count() + 2
+
+
+def test_derivative_wrt_constant():
+    mesh = UnitIntervalMesh(5)
+    V = FunctionSpace(mesh, "CG", 1)
+
+    u = TrialFunction(V)
+    v = TestFunction(V)
+    c = Constant(5)
+    f = Function(V)
+    solution = Function(V)
+
+    a = (c**2)*inner(u, v) * dx
+    L = inner(f, v) * dx
+    solve(a == L, solution)
+
+    d = derivative(a, c, IntValue(1))
+    solve(d == L, solution)
```

Comment:
assert solution1 == (c/2)*solution2 ?

Reply:
This is not a good idea, in general this is a hard problem to test the equality of two expressions symbolically: https://en.wikipedia.org/wiki/Computer_algebra#Equality

Reply:
`np.allclose(solution1.dat.data_ro, solution2.dat.data_ro)`?

Reply:
I did a thing